src/index.ts

```diff
@@ -67,14 +114,15 @@ const dmMachine = createMachine(
                 on: { SPEAK_COMPLETE: "HowCanIHelp" },
               },
               HowCanIHelp: {
-                entry: say("You can say whatever you like."),
+                entry: say("What genre of book are you interested in reading today?"),
                 on: { SPEAK_COMPLETE: "Ask" },
               },
               Ask: {
                 entry: listen(),
                 on: {
                   RECOGNISED: {
-                    target: "Repeat",
+                    guard: ({ event }) => getEntities(grammar[event.value[0].utterance].entities.genre, event.value[0].utterance),
```

Comment:
Is this what causes error? I guess the intended use of function `getEntities` is to provide it with entity and utterance, so the first argument should be just `"genre"` (assuming that the function works).

Reply:
yeah the function supposedly checks if any of the grammar entries, in this case genre, appears in the sentence/user's utterance. But I am getting uncaught errors about undefined entities or undefined genre when I check that in my guard.